diff --git a/template.ftl.orig b/template.ftl
index cc8d445..fe24771 100644
--- a/template.ftl.orig
+++ b/template.ftl
@@ -1,3 +1,11 @@
+<#function loginLinkMsg msgId>
+    <#local linkMsg = msg(msgId)?no_esc?markup_string>
+    <#if linkMsg == msgId>
+        <#local linkMsg = "">
+    </#if>
+    <#return linkMsg>
+</#function>
+
 <#import "footer.ftl" as loginFooter>
 <#macro registrationLayout bodyClass="" displayInfo=false displayMessage=true displayRequiredFields=false>
 <!DOCTYPE html>
@@ -25,6 +33,18 @@
             <link href="${url.resourcesPath}/${style}" rel="stylesheet" />
         </#list>
     </#if>
+    <#if properties.univentionCustomTheme?has_content>
+        <link rel="stylesheet" type="text/css" href="${properties.univentionCustomTheme}">
+    </#if>
+    <#if properties.univentionFavIcon?has_content>
+        <link rel="icon" href="${properties.univentionFavIcon}">
+    </#if>
+    <#if properties.univentionTheme?has_content>
+        <link rel="stylesheet" type="text/css" href="${properties.univentionTheme}">
+    </#if>
+    <#if properties.univentionCustomTheme?has_content>
+        <link rel="stylesheet" type="text/css" href="${properties.univentionCustomTheme}">
+    </#if>
     <#if properties.scripts?has_content>
         <#list properties.scripts?split(' ') as script>
             <script src="${url.resourcesPath}/${script}" type="text/javascript"></script>
@@ -62,6 +82,26 @@
 </head>
 
 <body class="${properties.kcBodyClass!}" data-page-id="login-${pageId}">
+<div class="cookie-banner hidden" style=".hidden { display: none; }">
+    <div class="modal-wrapper--isVisible modal-wrapper--isVisibleFullscreen">
+        <section class="dialog cookie-banner-modal">
+            <header class="dialog__header">
+                <h3>
+                    <span class="cookie-title" id="cookie-title">title</span>
+                </h3>
+            </header>
+            <main class="cookie-banner-text">
+                <div id="cookie-text">text</div>
+            </main>
+            <footer>
+                <button class="cookie-banner-button">
+                    <span>${msg("doAccept")}</span>
+                </button>
+            </footer>
+        </section>
+        <script>displayCookieBanner("${properties.univentionMetaJson}", "${locale.currentLanguageTag}");</script>
+    </div>
+</div>
 <div class="${properties.kcLoginClass!}">
     <div id="kc-header" class="${properties.kcHeaderClass!}">
         <div id="kc-header-wrapper"
@@ -176,6 +216,15 @@
 
       <@loginFooter.content/>
     </div>
+    <div id="umcLoginLinks">
+        <#list 1..12 as i>
+            <#assign linkHref = loginLinkMsg("linkHref" + i)?no_esc?markup_string>
+            <#assign linkDescription = loginLinkMsg("linkDescription" + i)?no_esc?markup_string>
+            <#if linkHref?has_content && linkDescription?has_content>
+                <a href="${linkHref}" target="_self">${linkDescription}</a>
+            </#if>
+        </#list>
+    </div>
   </div>
 </body>
 </html>
