--- template.ftl.orig	2023-08-22 10:13:34.625772537 +0200
+++ template.ftl	2023-08-22 10:14:02.665645137 +0200
@@ -34,6 +34,8 @@
             <script src="${script}" type="text/javascript"></script>
         </#list>
     </#if>
+    <link rel="stylesheet" type="text/css" href="${properties.univentionTheme}">
+    <link rel="stylesheet" type="text/css" href="${properties.univentionCommonTheme}">
 </head>

 <body class="${properties.kcBodyClass!}">
